src/providers/aws.rs

```diff
@@ -0,0 +1,887 @@
+//! AWS backend for tsunami.
+//!
+//! The primary `impl Launcher` type is [`Launcher`].
+//! It internally uses the lower-level, region-specific [`RegionLauncher`].
+//! Both these types use [`Setup`] as their descriptor type.
+//!
+//! This implementation uses [defined duration](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances)
+//! instances.
+//!
+//! # Example
+//! ```rust,no_run
+//! use tsunami::providers::{aws, Launcher};
+//! use tsunami::TsunamiBuilder;
+//!
+//! let mut b = TsunamiBuilder::default();
```

Comment:
More detail + comments plz. Also, it'd be great to include the one AWS-specific thing we have here (defined durations).

Reply:
The old main example is a good case-study:
https://github.com/jonhoo/tsunami/blob/96a2f912f445b57b02a217b4a6998837373989b6/src/lib.rs#L7-L38

Reply:
is it canonical to include the same example in multiple places throughout the docs?

Reply:
Arguably the top-level example should show _mixing_ AWS and Azure tsunamis, but barring that I think it's fine for them to be the same.

Reply:
sure, the mixing idea was relatively straightforward to write.